kernel/src/expressions/scalars.rs

```diff
@@ -3,33 +3,44 @@ use std::fmt::{Display, Formatter};
 
 use chrono::{DateTime, NaiveDate, NaiveDateTime, TimeZone, Utc};
 
-use crate::schema::{DataType, PrimitiveType};
+use crate::schema::{DataType, PrimitiveType, StructField};
 use crate::utils::require;
 use crate::{DeltaResult, Error};
 
 /// A single value, which can be null. Used for representing literal values
 /// in [Expressions][crate::expressions::Expression].
 #[derive(Debug, Clone, PartialEq)]
 pub enum Scalar {
+    /// 32bit integer
     Integer(i32),
+    /// 64bit integer
     Long(i64),
+    /// 16bit integer
     Short(i16),
+    /// 8bit integer
     Byte(i8),
+    /// 32bit floating point
     Float(f32),
+    /// 64bit floating point
     Double(f64),
     /// utf-8 encoded string.
     String(String),
     /// true or false value
     Boolean(bool),
     /// Microsecond precision timestamp, adjusted to UTC.
     Timestamp(i64),
-    /// Microsecond precision timestamp, without any timezone.
+    /// Microsecond precision timestamp, with no timezone.
     TimestampNtz(i64),
     /// Date stored as a signed 32bit int days since UNIX epoch 1970-01-01
     Date(i32),
+    /// Binary data
     Binary(Vec<u8>),
+    /// Decimal value with a given precision and scale.
     Decimal(i128, u8, u8),
+    /// Null value with a given data type.
     Null(DataType),
+    /// Struct value
+    Struct(Vec<Scalar>, Vec<StructField>),
```

Comment:
Hmm, it's annoying that we need to track value and type separately, but unavoidable because we need to know each field's name and nullability. But that means we need to validate somehow. At a minimum:
* same number of values and fields 
* data type of each value matches that of its field
* non-nullable fields should not have `Null` value

Not sure where to put the validation tho, since this is just an enum variant? Maybe we need the variant to take a proper struct with a `new` method?

We could also define a `ScalarStructField` or some such that combines `Scalar` with `StructField`... but that would probably be more annoying than it's worth?

Reply:
Meanwhile -- what is it like to actually create a struct literal in practice?
(I know it needs the two vec, but what kind of code experience is involved?)